### DISCUSSION TITLE
Duplicated commit ID in ProductVersion of NuGet DLLs

### DISCUSSION THREAD

**Discussion by KalleOlaviNiemitalo** [NuGet/Home#11895] [Q&A] [answered]
The ProductVersion field of the VERSIONINFO resource in NuGet DLLs contains two copies of the commit ID a8942f45835ba527cedacc02f7a75468f7a7b4bd. Is this by design or some kind of mistake?

```
PS C:\> Get-ChildItem -Path:"C:\Program Files\dotnet\sdk\6.0.301" -Filter:"NuGet*.dll" | Select-Object -Property Name,{$_.VersionInfo.ProductVersion}

Name                              $_.VersionInfo.ProductVersion
----                              -----------------------------
NuGet.Build.Tasks.Console.dll     6.2.1-rc.2+a8942f45835ba527cedacc02f7a75468f7a7b4bd.a8942f45835ba527cedacc02f7a75468f7a7b4bd
NuGet.Build.Tasks.dll             6.2.1-rc.2+a8942f45835ba527cedacc02f7a75468f7a7b4bd.a8942f45835ba527cedacc02f7a75468f7a7b4bd
NuGet.CommandLine.XPlat.dll       6.2.1-rc.2+a8942f45835ba527cedacc02f7a75468f7a7b4bd.a8942f45835ba527cedacc02f7a75468f7a7b4bd
NuGet.Commands.dll                6.2.1-rc.2+a8942f45835ba527cedacc02f7a75468f7a7b4bd.a8942f45835ba527cedacc02f7a75468f7a7b4bd
NuGet.Common.dll                  6.2.1-rc.2+a8942f45835ba527cedacc02f7a75468f7a7b4bd.a8942f45835ba527cedacc02f7a75468f7a7b4bd
NuGet.Configuration.dll           6.2.1-rc.2+a8942f45835ba527cedacc02f7a75468f7a7b4bd.a8942f45835ba527cedacc02f7a75468f7a7b4bd
NuGet.Credentials.dll             6.2.1-rc.2+a8942f45835ba527cedacc02f7a75468f7a7b4bd.a8942f45835ba527cedacc02f7a75468f7a7b4bd
NuGet.DependencyResolver.Core.dll 6.2.1-rc.2+a8942f45835ba527cedacc02f7a75468f7a7b4bd.a8942f45835ba527cedacc02f7a75468f7a7b4bd
NuGet.Frameworks.dll              6.2.1-rc.2+a8942f45835ba527cedacc02f7a75468f7a7b4bd.a8942f45835ba527cedacc02f7a75468f7a7b4bd
NuGet.LibraryModel.dll            6.2.1-rc.2+a8942f45835ba527cedacc02f7a75468f7a7b4bd.a8942f45835ba527cedacc02f7a75468f7a7b4bd
NuGet.Packaging.Core.dll          6.2.1-rc.2+a8942f45835ba527cedacc02f7a75468f7a7b4bd.a8942f45835ba527cedacc02f7a75468f7a7b4bd
NuGet.Packaging.dll               6.2.1-rc.2+a8942f45835ba527cedacc02f7a75468f7a7b4bd.a8942f45835ba527cedacc02f7a75468f7a7b4bd
NuGet.ProjectModel.dll            6.2.1-rc.2+a8942f45835ba527cedacc02f7a75468f7a7b4bd.a8942f45835ba527cedacc02f7a75468f7a7b4bd
NuGet.Protocol.dll                6.2.1-rc.2+a8942f45835ba527cedacc02f7a75468f7a7b4bd.a8942f45835ba527cedacc02f7a75468f7a7b4bd
NuGet.Versioning.dll              6.2.1-rc.2+a8942f45835ba527cedacc02f7a75468f7a7b4bd.a8942f45835ba527cedacc02f7a75468f7a7b4bd
```

I checked the version element of `NuGet.Common.nuspec` and the ProductInformationalVersionAttribute of `lib/netstandard2.0/NuGet.Common.dll` in a few older versions of the NuGet.Common package:

- <https://www.nuget.org/packages/NuGet.Common/6.2.1> has package version `6.2.1` and product version `6.2.1+a8942f45835ba527cedacc02f7a75468f7a7b4bd.a8942f45835ba527cedacc02f7a75468f7a7b4bd`.
- <https://www.nuget.org/packages/NuGet.Common/6.1.0> has package version `6.1.0` and product version `6.1.0+a8c1fff09f75c0291a6b91d8697a999ea5253825.a8c1fff09f75c0291a6b91d8697a999ea5253825`.
- <https://www.nuget.org/packages/NuGet.Common/6.0.0-preview.3> has package version `6.0.0-preview.3` and product version `6.0.0-preview.3+f017f82b310f26a97d8d8aafc7772d40121730a6.f017f82b310f26a97d8d8aafc7772d40121730a6`.
- <https://www.nuget.org/packages/NuGet.Common/5.9.1> has package version `5.9.1` and product version `5.9.1+d525b0e670f3b6cbd5c73a35f04730a9f658c852.d525b0e670f3b6cbd5c73a35f04730a9f658c852`.
- <https://www.nuget.org/packages/NuGet.Common/5.7.0> has package version `5.7.0+b804bf4ba62c0b47c77bbf3e22e196b57cd7a556` and product version `5.7.0+b804bf4ba62c0b47c77bbf3e22e196b57cd7a556.b804bf4ba62c0b47c77bbf3e22e196b57cd7a556`.
- <https://www.nuget.org/packages/NuGet.Common/5.3.0> has package version `5.3.0+b75150f2f4127a77a166c9552845e86fb24a3282` and product version `5.3.0+b75150f2f4127a77a166c9552845e86fb24a3282.b75150f2f4127a77a166c9552845e86fb24a3282`.
- <https://www.nuget.org/packages/NuGet.Common/5.3.0-rtm.6192> has package version `5.3.0-rtm.6192+bb60d6720d24890b8f3e071e70d27ea0f2bef57e` and product version `5.3.0-rtm.6192+bb60d6720d24890b8f3e071e70d27ea0f2bef57e.bb60d6720d24890b8f3e071e70d27ea0f2bef57e`.
- <https://www.nuget.org/packages/NuGet.Common/5.2.1> has package version `5.2.1+311b325795f23b3c718473671d3432cdbcf9deff` and product version `5.2.1+311b325795f23b3c718473671d3432cdbcf9deff`.
- <https://www.nuget.org/packages/NuGet.Common/5.2.0> has package version `5.2.0+f6f9fb481b9fbf14154d484dbf376446c9b0d274` and product version `5.2.0+f6f9fb481b9fbf14154d484dbf376446c9b0d274`.
- <https://www.nuget.org/packages/NuGet.Common/5.1.0> has package version `5.1.0+08414feaac77473f29f6ac182ab4c390ecd3542f` and product version `5.1.0+08414feaac77473f29f6ac182ab4c390ecd3542f`.
- <https://www.nuget.org/packages/NuGet.Common/5.0.0> has package version `5.0.0+42a8779499c1d1ed2488c2e6b9e2ee6ff6107766` and product version `5.0.0+42a8779499c1d1ed2488c2e6b9e2ee6ff6107766`.

Apparently, the format of the product version changed between 5.2.1 and 5.3.0-rtm.6192. This may have been caused by <https://github.com/NuGet/NuGet.Client/commit/4c70f115c1b418725a93f4d782573deae3646c2e> (in pull request <https://github.com/NuGet/NuGet.Client/pull/3002> to fix <https://github.com/NuGet/Home/issues/8449>) adding a reference to the Microsoft.SourceLink.GitHub package. If that [sets the SourceRevisionId property](https://github.com/dotnet/sourcelink/blob/51310e65e75010467993f793e1739d1a1dad50c5/src/Microsoft.Build.Tasks.Git/build/Microsoft.Build.Tasks.Git.targets#L32), then the [AddSourceRevisionToInformationalVersion target](https://github.com/dotnet/sdk/blob/2c798cbf6bf259ce49d94d91bc665a95d42d7a7f/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.GenerateAssemblyInfo.targets#L58-L69) appends it to the InformationalVersion property. NuGet.Client could disable that by setting IncludeSourceRevisionInInformationalVersion=false.

**Comment by zivkan** [comment, marked as answer]
Looks like a bug. Created issue that can be assigned in a sprint: https://github.com/NuGet/Home/issues/11974
